lib/cli/src/commands/run/wasi.rs

```diff
@@ -100,7 +101,7 @@ pub struct Wasi {
     ///
     /// Allows WASI modules to open TCP and UDP connections, create sockets, ...
     #[clap(long = "net")]
-    pub networking: bool,
+    pub networking: Option<Option<String>>,
```

Comment:
What is `Option<Option<String>>` supposed to mean?

This at the very least needs a comment and a doc comment explanation with examples.
I don't see how `Option<Option<>` has any meaning with clap also?

Reply:
The first `Option` will be initialized then only `--net` is specified: `Some(None)` and the inner `Option` will be initialized when `--net=<ruleset>` is specified: `Some(Some(ruleset))`.

I tried a bunch of other patterns but neither worked the same way. I can add the same explanation above to the code to make it clear.

Reply:
We decided to opt for a dedicated flag: `--net-filter` for the ruleset specification.